homeassistant/components/elkm1/__init__.py

```diff
@@ -175,14 +214,16 @@ def __init__(self, element, elk, elk_data):
         """Initialize the base of all Elk devices."""
         self._elk = elk
         self._element = element
+        self._prefix = elk_data['prefix']
         self._temperature_unit = elk_data['config']['temperature_unit']
-        self._unique_id = 'elkm1_{}'.format(
-            self._element.default_name('_').lower())
+        self._unique_id = 'elkm1_{prefix}_{name}'.format(
+            prefix=self._prefix,
+            name=self._element.default_name('_')).lower()
```

Comment:
If we modify the unique_id to lowercase, we might break the uniqueness of the prefix.

Reply:
ok, testing uniqueness on them as lowercase.  good catch.